app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -68,11 +68,22 @@
 
     private boolean batterySaver;
 
-    private boolean pokeFlyRunning = false;
     private int trainerLevel;
 
+    private Button launchButton;
+
     private final Point arcInit = new Point();
     private int arcRadius;
+    private final BroadcastReceiver pokeflyStateChanged = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            if (Pokefly.ACTION_START.equals(intent.getAction())) {
+                updateLaunchButtonText(true);
+            } else if (Pokefly.ACTION_STOP.equals(intent.getAction())) {
+                updateLaunchButtonText(false);
+            }
```

Comment:
Yes, we can. If we choose this option then the new Intent will need to be launched only after Pokefly has updated its _running_ variable value!